### PR TITLE
Issue/178 client transformations

### PR DESCRIPTION
**Pull request by jonase**

### CODE REVIEW

File: client/src/components/dataset/sidebars/ChangeDataType.jsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+              options={dataTypeOptions}
+              onChange={type => {
+                if (type === 'date') {
+                  this.mergeArgs({ newType: 'date', dateFormat: 'YYYY-MM-DD' });
+                }
+                this.mergeArgs({ newType: type });
+              }}
+            />
+          </div>
+          {newType === 'date' && column.type === 'text' ?
```
Comment (gabemart) [inline client/src/components/dataset/sidebars/ChangeDataType.jsx]: Apologies if this originally came from me, but what's the reason for checking the column type is text here?

File: client/src/components/dataset/DatasetTable.jsx [hunk 2]
```diff
@@ -114,136 +54,118 @@ export default class DatasetTable extends Component {
   }
 
   getCellClassName(columnTitle) {
-    if (this.state.activeSidebar &&
-      this.state.activeSidebar.columnTitle === columnTitle) {
+    const { sidebarProps } = this.state;
+    if (sidebarProps != null && sidebarProps.columnTitle === columnTitle) {
```
Comment (gabemart) [inline client/src/components/dataset/DatasetTable.jsx]: This works for "Filter" sidebar, but not for the change-data-type sidebar as that doesn't have the `columnTitle` property. For this to work correctly we need to also check if `sidebarProps.column.title === columnTitle`

File: client/src/components/dataset/sidebars/ChangeDataType.jsx (hunk 1 above)
Comment (jonase) [inline client/src/components/dataset/sidebars/ChangeDataType.jsx]: If the `column.type` is of type `number` this is a number->date transform. I will interpret the number as a timestamp so no date format needs to be specified. Perhaps we later add the ability to interpret the number as year, in which case you would here pick between timestamp or year. We haven't discussed this so that's why I don't do it.

TLDR; the available date format options (DD-MM-YYYY, MM-DD-YYYY, etc.) doesn't make sense when the column type in number

File: client/src/components/dataset/DatasetTable.jsx (hunk 2 above)
Comment (jonase) [inline client/src/components/dataset/DatasetTable.jsx]: 👍 I'll fix this. Also, we should compare columnName instead of title since we now have that property available

File: client/src/components/dataset/sidebars/ChangeDataType.jsx (hunk 1 above)
Comment (gabemart) [inline client/src/components/dataset/sidebars/ChangeDataType.jsx]: Ah I see - turns out I misunderstood what the date format option was for, I was viewing it as a display option rather than a parsing option. I can now see why this makes sense

Comment (jonase) [inline client/src/components/dataset/sidebars/ChangeDataType.jsx]: Yeah, it's a hint on how to parse the text. We store dates as timestamps
internally

**Comment by gabemart** [comment]
👍 LGTM
